### PR TITLE
Airwallex: Add `descriptor` field

### PR DESCRIPTION
**Pull request by dsmcclain**
CE-2530

Rubocop:
739 files inspected, no offenses detected

Unit:
5145 tests, 75476 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications

Remote:
Loaded suite test/remote/gateways/remote_airwallex_test
18 tests, 46 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications

### CODE REVIEW

File: test/unit/gateways/airwallex_test.rb [hunk 1]
```diff
@@ -179,15 +179,24 @@ def test_purchase_passes_request_id
     stub_comms do
       @gateway.purchase(@amount, @credit_card, @options.merge(request_id: request_id))
     end.check_request do |_endpoint, data, _headers|
-      assert_match(/request_/, data)
+      assert_match(request_id, data)
```
Comment (dsmcclain) [inline test/unit/gateways/airwallex_test.rb:182]: This change is unrelated to the `descriptor` field. Because the request contains a field named `request_id`, this assertion was not really testing the presence of the `request_{timestamp}` value.

File: test/unit/gateways/airwallex_test.rb [hunk 2]
```diff
@@ -179,15 +179,24 @@ def test_purchase_passes_request_id
...
     end.respond_with(successful_purchase_response)
   end
 
   def test_purchase_passes_currency_code
     stub_comms do
       @gateway.purchase(@amount, @credit_card, @options.merge(currency: 'USD'))
     end.check_request do |_endpoint, data, _headers|
-      assert_match(/request_/, data)
+      # only look for currency code on the create_payment_intent request
+      assert_match(/USD/, data) unless data.include?('_purchase')
```
Comment (dsmcclain) [inline test/unit/gateways/airwallex_test.rb:191]: This change is unrelated to the `descriptor` field. The previous assertion was not testing the currency code.

File: test/unit/gateways/airwallex_test.rb (hunk 1 above)
Comment (rachelkirk) [reply test/unit/gateways/airwallex_test.rb:182]: Nice, thanks for the description 😉

**Review by rachelkirk — APPROVED** [review]
Tests pass, nice fixes in the unit test. ✨ 🚢 ✨

File: lib/active_merchant/billing/gateways/airwallex.rb [hunk 3]
```diff
@@ -139,8 +136,8 @@ def create_payment_intent(money, options = {})
         post = {}
         add_invoice(post, money, options)
         add_order(post, options)
-        post[:request_id] = options[:request_id]
-        post[:merchant_order_id] = options[:merchant_order_id]
+        post[:request_id] = "#{request_id(options)}_setup"
+        post[:merchant_order_id] = "#{merchant_order_id(options)}_setup"
```
Comment (dsmcclain) [inline lib/active_merchant/billing/gateways/airwallex.rb:140]: Airwallex will fail a transaction if it uses a `request_id` or `merchant_order_id` that has been used in a previous transaction. Because their typical flow involves two transactions (the `create_payment_intent` followed by some sort of `purchase`), we have chosen to append this suffix to the ids on the `create_payment_intent` call in order to make them different than the subsequent `purchase` call.

Comment (rachelkirk) [reply lib/active_merchant/billing/gateways/airwallex.rb:140]: This is very readable code. Thanks for adding this explanation, I always learn something when reviewing your PR's.

**Review by rachelkirk — APPROVED** [review]
Tests pass, this looks great! 🚀
